rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -106,6 +111,24 @@ public SyncPeerStatus getPeer(Peer peer) {
     }
 
     public Optional<Peer> getBestPeer() {
+        if (topBest > 0.0D) {
+            List<Map.Entry<Peer, SyncPeerStatus>> entries = getBestCandidatesStream()
+                    .sorted(this.peerComparator.reversed())
+                    .collect(Collectors.toList());
+
+            if (entries.isEmpty()) {
+                return Optional.empty();
+            }
+
+            int numberOfPeersToConsider = (int) Math.round(((double) entries.size()) * (topBest / 100.0D));
```

Comment:
I'd double check edge cases here, eg. when this results to `zero` or to `entries.size()`. You may wanna use `Math.max(1, <...>)` and `Math.min(entries.size(), <...>)`. You may also want to validate `topBest` config value somewhere that it's in the `[0..100]` range - probably in `RskSystemProperties`

Reply:
Addressed